docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,328 @@
+| Follower `rollup-boost` Fails | Rollup Boost Failure | Follower sequencer `rollup-boost` fails. The leader sequencer is unaffected. Once `rollup-boost` is recovers, the sequencer will rejoin the set as a follower. |
+| Leader Builder Fails | Builder Failure| The builder associated with the sequencer leader fails and is no longer producing blocks. `rollup-boost` will fallback to the default execution client's payload and continue to produce blocks. Once the builder has been recovered and synced back to the chain tip, it will continue block production as normal.|
+| Follower Builder Fails| Builder Failure| The builder associated with a follower sequencer instance fails. Block production is not interrupted. Once the builder has been recovered it will sync back to the chain tip and process FCUs as normal.|
+| Leader Builder Producing Bad Blocks| Builder Failure| In this scenario, the builder is "healthy" but producing bad blocks (eg. empty blocks). If the builder block passes validation via a `new_payload` call to the default execution client, it will be proposed to the network. Manual intervention is needed to either switch to a different sequencer or shutoff the builder. Further mitigation can be introduced via block selection policy allowing `rollup-boost` to select the "healthiest" block. Currently, it is unclear what block selection policy would provide the strongest guarantees.|
+
+</details>
+
+### Pros and Cons
+
+<details>
+<summary>Pros/Cons</summary>
+
+#### Pros
+
+- Simple, easy to operate and monitor
+- No coordination required between `n` builders
+
+#### Cons
+
+- No builder failover/redundancy
+- Resource intensive for chains that do not want to run 3 builders
+
+</details>
+
+<br/>
+
+## `n` Block Builders
+
+In this design, each `rollup-boost` instance can be configured with `n` builders. When an FCU with payload attributes is received, it is broadcast to all builders, each of which begins building a block. Once `rollup-boost` receives a `get_payload` request from `op-node`, it multiplexes the request to all builders. Each builder returns its built payload, and if a payload isn't returned within `n` ms, it is ignored. A block selection policy is applied across the successful responses to choose the final payload (eg. selecting the block with the highest gas used). The selected payload is then validated with via a `new_payload` request sent to the sequencer's default execution client. In the event that the selected block is invalid, the next best block is selected and validated with the default execution client instead. If all builder blocks are invalid, then the default execution client's block is selected.
```

Comment:
Unfortunately this solution does not work mainly due to that we have to decide which builder's payload you're gonna potentially accept, otherwise, it's not feasible to decide who's flashblocks to stream to the public

Reply:
Yes we are aligned on this. After an initial round of feedback, the design doc has been updated to focus on 1:1 Rollup Boost <-> Builder deployments.